stan/math/rev/mat/fun/multiply.hpp

```diff
@@ -598,11 +608,21 @@ template <typename T1, int R1, int C1, typename T2,
           typename = require_any_var_t<T1, T2>>
 inline Eigen::Matrix<var, R1, C1> multiply(const Eigen::Matrix<T1, R1, C1>& m,
                                            const T2& c) {
-  // TODO(trangucci) pull out to eliminate overpromotion of one side
-  // move to matrix.hpp w. promotion?
   return to_var(m) * to_var(c);
 }
 
+template <typename T1, int R1, int C1, typename = require_arithmetic_t<T1>>
+inline Eigen::Matrix<var, R1, C1> multiply(const Eigen::Matrix<T1, R1, C1>& m,
+                                           const var& c) {
+  return m * c;
+}
+
+template <int R1, int C1, typename T2, typename = require_arithmetic_t<T2>>
+inline Eigen::Matrix<var, R1, C1> multiply(const Eigen::Matrix<var, R1, C1>& m,
+                                           const T2& c) {
```

Comment:
Did you try something like

```cpp
template <typename EigVar, typename Arith, 
   require_eigen_vt<is_var, EigVar>...,
   require_arithmetic_t<Arith>...>
inline Eigen::Matrix<var, R1, C1> multiply(EigVar&& m, Arith&& c) {
```

Reply:
I've tried
```
template <typename EigVar, typename Arith,
          require_eigen_vt<is_var, EigVar>..., require_arithmetic_t<Arith>...>
inline EigVar multiply(EigVar&& m, Arith&& c) {
```
and
```
template <typename EigVar, int R, int C, typename Arith,
          require_eigen_vt<is_var, EigVar>..., require_arithmetic_t<Arith>...>
inline Eigen::Matrix<var, R, C> multiply(EigVar&& m, Arith&& c) {
```
and it both cases this was not chosen over the unspecialised (preexisting) one, also if I specify `const` for the parameters.